src/popup/components/Header.vue

```diff
@@ -215,6 +244,10 @@ export default defineComponent({
   z-index: variables.$z-index-header;
   height: var(--header-height);
 
+  &.hidden {
+    z-index: -1;
+  }
```

Comment:
Are you sure using only the `z-index` is OK? Looks rather off in my opinion. Maybe adding opacity/visibility change would be good?